grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -113,6 +151,462 @@ class OrganismController {
+            , @RestApiParam(name = "genus", type = "string", paramType = RestApiParamType.QUERY, description = "species genus")
+            , @RestApiParam(name = "blatdb", type = "string", paramType = RestApiParamType.QUERY, description = "filesystem path for a BLAT database (e.g. a .2bit file)")
+            , @RestApiParam(name = "publicMode", type = "boolean", paramType = RestApiParamType.QUERY, description = "a flag for whether the organism appears as in the public genomes list")
+            , @RestApiParam(name = "commonName", type = "string", paramType = RestApiParamType.QUERY, description = "a name used for the organism")
+            , @RestApiParam(name = "nonDefaultTranslationTable", type = "string", paramType = RestApiParamType.QUERY, description = "non-default translation table")
+            , @RestApiParam(name = "metadata", type = "string", paramType = RestApiParamType.QUERY, description = "organism metadata")
+            , @RestApiParam(name = "organismData", type = "file", paramType = RestApiParamType.QUERY, description = "ZIP compressed data directory")
+    ])
+    @Transactional
+    def addOrganismWithSequence() {
+        /*
+            Representative CURL query:
+
+            curl http://localhost:8080/apollo/organism/addOrganismWithSequence  \
+                -F "commonName=Amel"
+                -F "username=admin"
+                -F "password=admin"
+                -F "organismData=@/path/to/compressed-jbrowse-data.zip"
+                -X POST
+                -v
+         */
+        log.debug "addOrganismWithSequence with params: ${params.toString()}"
+        JSONObject returnObject = new JSONObject()
+        returnObject.put("status", new JSONArray())
+        JSONObject requestObject = permissionService.handleInput(request, params)
+        String clientToken = requestObject.getString(FeatureStringEnum.CLIENT_TOKEN.value)
+
+        if (!requestObject.containsKey("commonName")) {
+            returnObject.getJSONArray("status").add("Error: Organism commonName not provided.")
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST)
+            render returnObject as JSON
+            return
+        }
+
+        if (!requestObject.containsKey("directory")) {
+            if (request.getFile("organismData") == null) {
+                returnObject.getJSONArray("status").add("Error: Neither 'directory' nor 'organismData' provided. /addOrganismWithSequence requires either one of them.")
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST)
+                render returnObject as JSON
+                return
+            }
+        }
+
+        String directoryName
+
+        try {
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(requestObject))) {
+                log.debug "User is admin"
+                def existingOrganism
+
+                if (requestObject.containsKey("directory")) {
```

Comment:
confused, would this not just duplicate the other addOrganism method?  Would they just not use that method instead?